document/src/tests/documentselectparsertest.cpp

```diff
@@ -137,11 +137,11 @@ DocumentUpdate::SP DocumentSelectParserTest::createUpdate(
         const std::string& hstr)
 {
     const DocumentType* type = _repo->getDocumentType(doctype);
-    DocumentUpdate::SP doc(new DocumentUpdate(*_repo, *type, DocumentId(id)));
-    doc->addUpdate(FieldUpdate(doc->getType().getField("headerval"))
-                      .addUpdate(AssignValueUpdate(IntFieldValue(hint))));
-    doc->addUpdate(FieldUpdate(doc->getType().getField("hstringval"))
-                      .addUpdate(AssignValueUpdate(StringFieldValue(hstr))));
+    auto doc = std::make_shared<DocumentUpdate>(*_repo, *type, DocumentId(id));
+    doc->addUpdate(std::move(FieldUpdate(doc->getType().getField("headerval"))
```

Comment:
I think most (if not all?) of these explicit moves for chained calls can be removed; see comment for rvalue `addUpdate` overload

Reply:
Indeed, gone